Add new llm providers (#38)
ITMO-NSS-team/Open-Source-Advisor: README.md

<p align="center">
	<img src="https://img.shields.io/badge/tqdm-FFC107.svg?style=BadgeStyleOptions.DEFAULT&logo=tqdm&logoColor=black"alt="tqdm">
	<img src="https://img.shields.io/badge/Pytest-0A9EDC.svg?style=BadgeStyleOptions.DEFAULT&logo=Pytest&logoColor=white"alt="Pytest">
	<img src="https://img.shields.io/badge/Docker-2496ED.svg?style=BadgeStyleOptions.DEFAULT&logo=Docker&logoColor=white"alt="Docker">
	<img src="https://img.shields.io/badge/Python-3776AB.svg?style=BadgeStyleOptions.DEFAULT&logo=Python&logoColor=white"alt="Python">
	<img src="https://img.shields.io/badge/GitHub%20Actions-2088FF.svg?style=BadgeStyleOptions.DEFAULT&logo=GitHub-Actions&logoColor=white"alt="GitHub%20Actions">
	<img src="https://img.shields.io/badge/AIOHTTP-2C5BB4.svg?style=BadgeStyleOptions.DEFAULT&logo=AIOHTTP&logoColor=white"alt="AIOHTTP">
	<img src="https://img.shields.io/badge/OpenAI-412991.svg?style=BadgeStyleOptions.DEFAULT&logo=OpenAI&logoColor=white"alt="OpenAI">
	<img src="https://img.shields.io/badge/Pydantic-E92063.svg?style=BadgeStyleOptions.DEFAULT&logo=Pydantic&logoColor=white"alt="Pydantic">
</p>
<br>

---
## Overview

<overview>

> Open-Source-Advisor is a productivity tool designed to simplify the management of GitHub repositories by automating tasks related to documentation and configuration. It enhances user experience through customizable outputs, ensuring that project documentation is always current. This tool is ideal for developers seeking to streamline their workflows and maintain high-quality documentation effortlessly.

Reviewer: Тоже кажется почти все слова не по делу.

</overview>

---

## Table of contents

- [Core features](#core-features)
- [Installation](#installation)
- [Getting started](#getting-started)
- [Examples](#examples)
- [Contributing](#contributing)
- [License](#license)
- [Acknowledgments](#acknowledgments)
- [Citation](#citation)

---

> ## Core features

Reviewer 2: Этот раздел стоит переписать без воды: что реально умеет ОСА? Чтобы юзеру было понятно, зачем её использовать
Генерация ридми, документации, настройка CI/CD, автоматическое создание PR со всеми изменениями и всё такое

<corefeatures>

1. **Main Application Logic**: Centralized in `main.py`, orchestrating overall system behavior.

2. **Configuration Management**: Utilizes multiple TOML files for flexible parameter adjustments and prompts.

3. **Containerization Support**: Dockerfile ensures consistent environments across platforms for deployment.

4. **GitHub Integration**: Automates tasks via `githubagent.py`, enhancing collaboration and version control.

5. **Documentation Generation**: Parses and generates up-to-date project documentation using dedicated scripts.

</corefeatures>

---

## Installation

Install Open-Source-Advisor using one of the following methods:

**Build from source:**

1. Clone the Open-Source-Advisor repository:
```sh
❯ git clone https://github.com/ITMO-NSS-team/Open-Source-Advisor
```

2. Navigate to the project directory:
```sh
❯ cd Open-Source-Advisor
```

3. Install the project dependencies:

> **Using `pip`** &nbsp;

Reviewer 2: Можно ещё добавить про установку через poetry

[<img align="center" src="https://img.shields.io/badge/Pip-3776AB.svg?style={badge_style}&logo=pypi&logoColor=white" />](https://pypi.org/project/pip/)

```sh
❯ pip install -r requirements.txt
```

**Using `docker`** &nbsp;
[<img align="center" src="https://img.shields.io/badge/Docker-2CA5E0.svg?style={badge_style}&logo=docker&logoColor=white" />](https://www.docker.com/)

```sh
❯ docker build -f docker/Dockerfile -t {image-name} .
```

---

## Getting started

### Prerequisites

OSA requires Python 3.10 or higher.

> File `.env` is required to specify GitHub token and LLM API key (OPENAI_API_KEY or VSE_GPT_KEY)

Reviewer 2: Название гитхаб-токена тоже стоит указать

Local Llama ITMO:
```sh
python main.py https://github.com/ITMO-NSS-team/Open-Source-Advisor
```  
OpenAI:
```sh
python main.py https://github.com/ITMO-NSS-team/Open-Source-Advisor openai gpt-3.5-turbo
```
VseGPT:
```sh
python main.py https://github.com/ITMO-NSS-team/Open-Source-Advisor vsegpt openai/gpt-3.5-turbo
```

---

## Contributing

- **[Report Issues](https://github.com/ITMO-NSS-team/Open-Source-Advisor/issues )**: Submit bugs found or log feature requests for the Open-Source-Advisor project.

---

## License

This project is protected under the BSD 3-Clause "New" or "Revised" License. For more details, refer to the [LICENSE](https://github.com/ITMO-NSS-team/Open-Source-Advisor/blob/main/LICENSE) file.

---

## Acknowledgments

- [**Open-source-ops**](https://github.com/aimclub/open-source-ops)

- [**Readme-ai**](https://github.com/eli64s/readme-ai)

---

## Citation

If you use this software, please cite it as below.

### APA format:

    ITMO-NSS-team (2024). Open-Source-Advisor repository [Computer software]. https://github.com/ITMO-NSS-team/Open-Source-Advisor

### BibTeX format:

    @misc{Open-Source-Advisor,

>         author = {ITMO-NSS-team},

Reviewer 2: Возник вопрос, как указывать авторов
@nicl-nno подскажешь?

Reviewer: Да можно написать NSS Lab, вроде для опенсорса редко авторов пречисляют.

	<img src="https://img.shields.io/badge/GitHub%20Actions-2088FF.svg?style=BadgeStyleOptions.DEFAULT&logo=GitHub-Actions&logoColor=white"alt="GitHub%20Actions">
	<img src="https://img.shields.io/badge/AIOHTTP-2C5BB4.svg?style=BadgeStyleOptions.DEFAULT&logo=AIOHTTP&logoColor=white"alt="AIOHTTP">
	<img src="https://img.shields.io/badge/OpenAI-412991.svg?style=BadgeStyleOptions.DEFAULT&logo=OpenAI&logoColor=white"alt="OpenAI">
	<img src="https://img.shields.io/badge/Pydantic-E92063.svg?style=BadgeStyleOptions.DEFAULT&logo=Pydantic&logoColor=white"alt="Pydantic">
</p>
<br>

---
## Overview

<overview>
Open-Source-Advisor is an open source Python library that provides a one-click way to get recommendations for improving an open source repository; generate documentation and code to implement those recommendations.
</overview>

---

## Table of contents

- [Core features](#core-features)
- [Installation](#installation)
- [Getting started](#getting-started)
- [Examples](#examples)
- [Contributing](#contributing)
- [License](#license)
- [Acknowledgments](#acknowledgments)
- [Citation](#citation)

---

## Core features

<corefeatures>

> 1. **Generation README file**: Automates the creation of a clear and structured README file for a repository, including projects with research papers.

Reviewer 2: **README file generation**: Automates the creation of a clear and structured README file for a repository, including projects based on research papers.

> 2. **Generation Documentation**: Automatically generates docstrings for Python code.

Reviewer 2: **Documentation generation**: Automatically generates docstrings for Python code.

> 3. **Automatic implementation of changes**: Clones the repository, creates a branch, commits and pushes changes, and creates a pull request.

Reviewer 2: and creates a pull request with proposed changes.

> 4. **Deployed in ITMO servers**: Use OSA locally or hosted and running version on ITMO servers.

Reviewer 2: **Deployed on ITMO servers**: Use OSA via API (OpenAI, VseGPT), on a local server, or a version hosted on ITMO servers.

# Open-Source-Advisor
Tool that just makes your open source project better

### How to use readme generator:

> You need to pass **URL of the GitHub repository**, **LLM API service provider** and **Specific LLM model to use**

Reviewer 3: I do not like the word *need* here

```suggestion
**URL of the GitHub repository**, **LLM API service provider** and **Specific LLM model to use** are required to use the generator.
```

3. **Containerization Support**: Dockerfile ensures consistent environments across platforms for deployment.

4. **GitHub Integration**: Automates tasks via `githubagent.py`, enhancing collaboration and version control.

5. **Documentation Generation**: `docgen.py` creates user-friendly guides, improving accessibility and user experience.

</corefeatures>

---

## Installation

Install Open-Source-Advisor using one of the following methods:

**Build from source:**

1. Clone the Open-Source-Advisor repository:
```sh
❯ git clone https://github.com/ITMO-NSS-team/Open-Source-Advisor
```

2. Navigate to the project directory:
```sh
❯ cd Open-Source-Advisor
```

3. Install the project dependencies:

**Using `pip`** &nbsp;
[<img align="center" src="https://img.shields.io/badge/Pip-3776AB.svg?style={badge_style}&logo=pypi&logoColor=white" />](https://pypi.org/project/pip/)

```sh
❯ pip install -r requirements.txt
```

**Using `docker`** &nbsp;
[<img align="center" src="https://img.shields.io/badge/Docker-2CA5E0.svg?style={badge_style}&logo=docker&logoColor=white" />](https://www.docker.com/)

```sh

> ❯ docker build -t ITMO-NSS-team/Open-Source-Advisor .

Reviewer 4: ```suggestion
❯ docker build -f docker/Dockerfile -t {image-name} .
```
Также можно загрузить образ в Docker Hub и в дальнейшем предлагать использовать docker pull.

Reviewer 4: На данный момент в планах добавить docker-compose.yaml, чтобы при запуске контейнера не нужно было прописывать docker run --env-file .env, поэтому итоговый текст может снова измениться.
